File: src/main/java/com/henryinqz/shopifybackendinternchallengef22/model/Location.java
package com.henryinqz.shopifybackendinternchallengef22.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("location")
public class Location {
    @Id
    public String id;

    public String name;
    public String address;
    public String zipCode;
    public String city;
    public String state;
    public String country;

    public Location(String name, String address, String zipCode, String city, String state, String country) {
        this.name = name;
        this.address = address;
        this.zipCode = zipCode;
        this.city = city;
        this.state = state;
        this.country= country;
    }
}

File: src/main/resources/static/js/location.js
const BASE_LOCATIONS_ENDPOINT = "/api/locations"

function onLocationSubmit() {
    const name = document.querySelector("#createLocationName").value
    const address = document.querySelector("#createLocationAddress").value
    const zipCode = document.querySelector("#createLocationZipCode").value
    const city = document.querySelector("#createLocationCity").value
    const state = document.querySelector("#createLocationState").value
    const country = document.querySelector("#createLocationCountry").value

    if (name !== '' && address !== '' && zipCode !== '' && city !== '' && state !== '' && country !== '') {
        fetch(`${BASE_LOCATIONS_ENDPOINT}/create`, {
            method: "POST",
            headers: {"content-type": "application/json"},
            body: JSON.stringify({
                name: name,
                address: address,
                zipCode: zipCode,
                city: city,
                state: state,
                country: country
            })
        })
            .then(response => response.json())
            .then(() => window.location.reload())
    }
}

function onLocationSubmitUpdate(id) {
    fetch(`${BASE_LOCATIONS_ENDPOINT}/${id}`, {
        method: "PUT",
        headers: {"content-type": "application/json"},
        body: JSON.stringify({
            name: "Updated location name!",
            address: "123 Happyjack Street",
            zipCode: "H0H0H0",
            city: "Markham",
            state: "ON",
            country: "Canada"
        })
    })
        .then(response => response.json())
        .then(data => console.log(data))
        .then(() => window.location.reload())
}

function onLocationDelete(id) {
    fetch(`${BASE_LOCATIONS_ENDPOINT}/${id}`, {
        method: "DELETE"
    })
        .then(response => response.json())
        .then(() => window.location.reload())
}